### PR TITLE
Add watchOS, tvOS and macOS targets

### PR DESCRIPTION
**Pull request by CherryPerry**
Supported platforms:
- iosArm32
- iosArm64
- iosX64
- watchosArm32
- watchosArm64
- watchosX86
- tvosArm64
- tvosX64
- macosX64

Tests check `iosX64`, `watchosX86`, `tvosX64`, `macosX64`.
No samples.

Fixes https://github.com/badoo/Reaktive/issues/342

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -108,6 +108,24 @@ void setupAllTargetsWithDefaultSourceSets(Project project, Boolean isLinuxArm32H
...
+                watchosSimMain.dependsOn iosCommonMain
+                watchosSimTest.dependsOn iosCommonTest
+
+                tvosArm64Main.dependsOn iosCommonMain
+                tvosArm64Test.dependsOn iosCommonTest
+
+                tvosSimMain.dependsOn iosCommonMain
+                tvosSimTest.dependsOn iosCommonTest
+
+                macosX64Main.dependsOn iosCommonMain
```
Comment (arkivanov) [inline build.gradle]: Why macos main depends on iosCommon? Maybe rename `iosCommonXxx` to `darwinCommonXxx`?

File: build.gradle [hunk 2]
```diff
@@ -206,6 +224,6 @@ import tasks.BuildIosSampleTask
 
 if (Target.currentTarget().isIos()) {
```
Comment (arkivanov) [inline build.gradle]: Is it really ios only or the whole darwin family?

**Comment by arkivanov** [comment]
> Tests check `iosX64`, `watchosX86` and `tvosX64`.

What about tests for `macos`?

**Comment by CherryPerry** [comment]
@arkivanov MacOS test are available out of box, updated description.

File: build.gradle [hunk 3]
```diff
@@ -199,13 +217,13 @@ private void setupLinuxArm32HfpTarget(Project project) {
...
+    project.apply plugin: DarwinPlugin
 }
 
 import tasks.BuildIosSampleTask
 
-if (Target.currentTarget().isIos()) {
+if (Target.currentTarget().isDarwin()) {
     tasks.register('buildIosSample', BuildIosSampleTask) {
-        dependsOn ':sample-mpp-module:fatDebug', ':sample-mpp-module:fatRelease'
+        dependsOn ':sample-mpp-module:fatIosDebug', ':sample-mpp-module:fatIosRelease'
```
Comment (arkivanov) [inline build.gradle]: It should be still named as "ios" right?

Comment (CherryPerry) [reply build.gradle]: Yes, sample has iOS version only now.

**Pull request merged**